rest/config_manager.go

```diff
@@ -761,56 +761,56 @@ func (b *bootstrapContext) ComputeMetadataIDForDbConfig(ctx context.Context, con
 //  4. The _default._default collection isn't already associated with a different metadata ID (syncInfo document is not present, or has a value of defaultMetadataID)
 func (b *bootstrapContext) computeMetadataID(ctx context.Context, registry *GatewayRegistry, config *DbConfig) string {
 
-	standardMetadataID := b.standardMetadataID(config.Name)
-
-	// If there's already a metadataID assigned to this database in the registry (including other config groups), use that
-	defaultMetadataIDInUse := false
-	for _, cg := range registry.ConfigGroups {
-		for dbName, db := range cg.Databases {
-			if dbName == config.Name {
-				return db.MetadataID
-			}
-			if db.MetadataID == defaultMetadataID {
-				defaultMetadataIDInUse = true
+	metadataID, source := func() (string, string) {
+		standardMetadataID := b.standardMetadataID(config.Name)
+
+		// If there's already a metadataID assigned to this database in the registry (including other config groups), use that
+		for _, cg := range registry.ConfigGroups {
+			for dbName, db := range cg.Databases {
+				if dbName == config.Name {
+					return db.MetadataID, fmt.Sprintf("Using metadata ID=%q from registry for db %q", db.MetadataID, dbName)
+				}
+				if db.MetadataID == defaultMetadataID {
+					// If the default metadata ID is already in use in the registry by a different database, use standard ID.
+					return standardMetadataID, fmt.Sprintf("Using metadata ID %q for db %q since db %q is using the default metadata ID", standardMetadataID, config.Name, dbName)
```

Comment:
This changes the previous behaviour in the case where:
- the default metadata ID is in use by a different db
- there is already a metadataID assigned to this database, and it's not the standardMetadataID

This is the reason for the previous defaultMetadataIDInUse approach, and I don't think this PR should change that functional behaviour.